lib/model/CProbabilityAndInfluenceCalculator.cc

```diff
@@ -827,6 +859,42 @@ bool CProbabilityAndInfluenceCalculator::calculate(double& probability) const {
     return m_Probability.calculate(probability);
 }
 
+bool CProbabilityAndInfluenceCalculator::calculateExplainingProbabilities(
+    TStrDoubleUMap& explainingProbabilities) const {
+
+    double probability{0.0};
+    for (const auto& ep : m_ExplainingProbabilities) {
+        if (!ep.second.calculate(probability)) {
+            return false;
+        } else {
+            explainingProbabilities.emplace(ep.first, probability);
+        }
+    }
+
+    return true;
+}
+
+bool CProbabilityAndInfluenceCalculator::calculateMultiBucketImpact(double& multiBucketImpact) const {
+    TStrDoubleUMap explainingProbabilities;
+    if (!this->calculateExplainingProbabilities(explainingProbabilities)) {
+        LOG_INFO(<< "Failed to compute explaining probabilities");
+        return false;
+    } else {
+        double sbProbability = explainingProbabilities[SINGLE_BUCKET_FEATURE_LABEL];
+        double mbProbability = explainingProbabilities[MULTI_BUCKET_FEATURE_LABEL];
+
+        double ls = std::log(std::max(sbProbability, ml::maths::MINUSCULE_PROBABILITY));
+        double lm = std::log(mbProbability);
```

Comment:
This should be bounded as well, i.e. `std::log(std::max(mbProbability, ...))`, in case of underflow. Also these should use `CTools::smallestProbability()`.